### PR TITLE
Pixi for installation and execution

### PR DESCRIPTION
**Pull request by fangohr**
Here is a pull request that introduces pixi for installation of the python tool. I have also added the `tasks`  to the `pixi.toml` file, and documented this in the `src/README.md`.

Perhaps @santaep or @HoWilgh can try if this works if they follow the instructions. 

If you are happy, do merge.

### CODE REVIEW

File: pixi.toml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+channels = ["conda-forge"]
+description = "Add a short description here"
+name = "MagneticMaterialsOntology"
+platforms = ["win-64", "linux-64", "osx-64", "osx-arm64"]
+version = "0.1.0"
+
+[tasks]
+build = "python src/build_onto.py"
+check = "emmocheck magnetic_material_mammos.ttl"
+docs = "ontodoc --format simple-html magnetic_material_mammos.ttl doc/magnetic_material_mammos.html"
```
Comment (samjrholt) [inline pixi.toml:12]: Running `pixi run docs` gives an error 

```
pixi run docs
✨ Pixi task (docs): ontodoc --format simple-html magnetic_material_mammos.ttl doc/magnetic_material_mammos.html
Traceback (most recent call last):
  File "/home/sam/repos/MagneticMaterialsOntology/.pixi/envs/default/bin/ontodoc", line 293, in <module>
    main()
    ~~~~^^
  File "/home/sam/repos/MagneticMaterialsOntology/.pixi/envs/default/bin/ontodoc", line 278, in main
    docpp.process()
    ~~~~~~~~~~~~~^^
  File "/home/sam/repos/MagneticMaterialsOntology/.pixi/envs/default/lib/python3.13/site-packages/ontopy/ontodoc.py", line 1138, in process
    self.process_alls()
    ~~~~~~~~~~~~~~~~~^^
  File "/home/sam/repos/MagneticMaterialsOntology/.pixi/envs/default/lib/python3.13/site-packages/ontopy/ontodoc.py", line 1017, in process_alls
    self.lines[i:i] = self.ontodoc.itemsdoc(
                      ~~~~~~~~~~~~~~~~~~~~~^
        items, int(opts.header_level)  # pylint: disable=no-member
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ).split("\n")
    ^
  File "/home/sam/repos/MagneticMaterialsOntology/.pixi/envs/default/lib/python3.13/site-packages/ontopy/ontodoc.py", line 451, in itemsdoc
    doc.append(self.itemdoc(item, header_level))
               ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "/home/sam/repos/MagneticMaterialsOntology/.pixi/envs/default/lib/python3.13/site-packages/ontopy/ontodoc.py", line 340, in itemdoc
    point=asstring(prop, link_style, ontology=onto),
          ~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sam/repos/MagneticMaterialsOntology/.pixi/envs/default/lib/python3.13/site-packages/ontopy/utils.py", line 262, in asstring
    string = asstring(
        expr.value, link, recursion_depth + 1, ontology=ontology
    )
  File "/home/sam/repos/MagneticMaterialsOntology/.pixi/envs/default/lib/python3.13/site-packages/ontopy/utils.py", line 314, in asstring
    raise RuntimeError(f"Unknown expression: {expr!r} (type: {type(expr)!r})")
RuntimeError: Unknown expression: None (type: <class 'NoneType'>)
```

File: src/README.md [hunk 2]
```diff
@@ -31,6 +31,27 @@ or simply run
 which builds the ontology, runs the emmocheck and creates a documentation. Please set the relevant path to `emmocheck` and `ontodoc` in the bash script `create.sh`
 
 
+### Installation and execution using pixi
+
+1. install pixi from https://pixi.sh/
+
+2. clone this repository:
+
+   git clone https://github.com/MaMMoS-project/MagneticMaterialsOntology.git
```
Comment (samjrholt) [inline src/README.md:40]:
```suggestion
   `git clone https://github.com/MaMMoS-project/MagneticMaterialsOntology.git`
```

**Review by samjrholt — CHANGES_REQUESTED** [review]
Looks good. I've successfully installed and run the first two tasks but get an error running the `docs` task

**Comment by thomasschrefl** [comment]
> Looks good. I've successfully installed and run the first two tasks but get an error running the `docs` task

I do not where this error comes from. By the way in the meantime, I changed doc section of the create.sh script.
We should also update the pixi task accordlingy.

**Comment by fangohr** [comment]
I fixed the error that Tom saw by merging the changes on the `main` branch into this PR.

**Comment by fangohr** [comment]
This is ready to be merged, I think.

**Comment by fangohr** [comment]
We still have an error that occurs, but that has nothing todo with the addition of the pixi-based software installation or task execution: it also occurs on main when running the `create.sh` script.

With this merge request , we can and will do the build  (equivalent to running `create.sh`) for every change to the repository, and that will tell us early if something is broken. 

I have added a "badge" like this 

[build](https://github.com/MaMMoS-project/MagneticMaterialsOntology/actions/workflows/build.yaml)

to the README (at the very bottom for now) to quickly see the status.

@samjrholt are you happy for this to be merged?

**Review by samjrholt — APPROVED** [review]

**Pull request merged**